Assignment4/sofiaferben/task07.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1FH_-DJcfynrnNfTPZqiwd1u4SkhtbGa8
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""First let's read the RDF file"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+ns = Namespace("http://somewhere#")
+##1) RDFLib
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+  print("Subclass of LivingThing:", s)
+
+##2) SPARQL
+from rdflib.plugins.sparql import prepareQuery
+q1 = prepareQuery('''
+SELECT ?sub
+WHERE {
+  ?sub rdfs:subClassOf ns:LivingThing.
+      }
+      ''',
+      initNs={"rdfs":RDFS, "ns":ns})
+# Visualize the results
+print("\nSubclass of  LivingThing")
+for r in g.query(q1):
+  print(r.sub)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+##1) RDFLib
+print("List of all individuals of Person (with subclasses)")
+for s,p,o in g.triples((None, RDF.type, ns.Person)):
+    print(s)
+
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.Person)):
+    for s1,p2,o2 in g.triples((None, RDF.type, s)):
+        print(s1)
+
+
+
+##2) SPARQL
+q2 = prepareQuery('''
+  SELECT DISTINCT ?s
+  WHERE {
+    {?s rdf:type ns:Person}
+    UNION
+    {
+     ?sub rdfs:subClassOf* ns:Person .
+     ?s  rdf:type ?sub
+    }
+  }
+  ''',
+  initNs = { "ns": ns, "rdf": RDF}
+  )
+
+print("\nList of all individuals of Person (with subclasses)")
+for r in g.query(q2):
+  print(r.s)
+
+"""**TASK 7.3: List all individuals of "Person" or "Animal" and all their properties including their class with RDFLib and SPARQL. You do not need to list the individuals of the subclasses of person**
+
+"""
+
+##1) RDFLib
+print("List of invidividuals of Person and their properties:")
+for s, p, o in g.triples((None, RDF.type, ns.Person)):
+  for sub, pre, obj in g.triples((s, None, None)):
+    print(sub, obj)
+
+print("\nList of invidividuals of Animal and their properties:")
+for s, p, o in g.triples((None, RDF.type, ns.Animal)):
+  for sub, pre, obj in g.triples((s, None, None)):
+    print(sub, obj)
```

Comment:
you are not listing the properties of the individuals, which is what the exercise requires